RealmSwift/Projection.swift

```diff
@@ -0,0 +1,642 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+import Realm.Private
+#if canImport(Combine)
+import Combine
+#endif
+
+private protocol AnyProjected {
+    var projectedKeyPath: AnyKeyPath { get }
+}
+
+// MARK: Projection
+
+/// ``@Projected`` is used to declare properties on ``Projection`` protocols which should be
+/// managed by Realm.
+///
+/// Example of usage:
+/// ```swift
+/// public class Person: Object {
+///     @Persisted var firstName = ""
+///     @Persisted var lastName = ""
+///     @Persisted var address: Address? = nil
+///     @Persisted var friends = List<Person>()
+///     @Persisted var reviews = List<String>()
+/// }
+///
+/// class PersonProjection: Projection<Person> {
+///     @Projected(\Person.firstName) var firstName
+///     @Projected(\Person.lastName.localizedUppercase) var lastNameCaps
+///     @Projected(\Person.address.city) var homeCity
+///     @Projected(\Person.friends.projectTo.firstName) var firstFriendsName: ProjectedCollection<String>
+/// }
+///
+/// let people: Results<PersonProjection> = realm.objects(PersonProjection.self)
+/// ```
+@propertyWrapper
+public struct Projected<T: ObjectBase, Value>: AnyProjected {
+
+    fileprivate var _projectedKeyPath: KeyPath<T, Value>!
```

Comment:
This doesn't appear to need to be a optional.

Reply:
fixed